torch/_inductor/ir.py

```diff
@@ -7307,6 +7308,12 @@ def codegen(self, wrapper):
         if isinstance(self.layout, Layout):
             self.codegen_size_asserts(wrapper)
 
+    def get_mutation_names(self):
+        return [self.inputs[self.idx_for_inplace_sum].get_name()]
```

Comment:
Since post op `sum` and `binary add` reusing same IR and inplace mutation is only valid for post op `sum`, I think we should check it here for branch.